specs/deneb/p2p-interface.md

```diff
@@ -95,10 +86,39 @@ class BlobIdentifier(Container):
 ##### `verify_blob_sidecar_signature`
 
 ```python
-def verify_blob_sidecar_signature(state: BeaconState, signed_blob_sidecar: SignedBlobSidecar) -> bool:
-    proposer = state.validators[signed_blob_sidecar.message.proposer_index]
-    signing_root = compute_signing_root(signed_blob_sidecar.message, get_domain(state, DOMAIN_BLOB_SIDECAR))
-    return bls.Verify(proposer.pubkey, signing_root, signed_blob_sidecar.signature)
+def verify_blob_sidecar_signature(state: BeaconState, blob_sidecar: BlobSidecar) -> bool:
+    block_header = blob_sidecar.signed_block_header.message
+    proposer = state.validators[block_header.proposer_index]
+    domain = get_domain(state, DOMAIN_BEACON_PROPOSER, compute_epoch_at_slot(block_header.slot))
+    signing_root = compute_signing_root(block_header, domain)
+    return bls.Verify(proposer.pubkey, signing_root, blob_sidecar.signed_block_header.signature)
+```
+
+##### `verify_blob_sidecar_inclusion_proof`
+
+```python
+def verify_blob_sidecar_inclusion_proof(blob_sidecar: BlobSidecar) -> bool:
+    commitment_item_gindex = MAX_BLOB_COMMITMENTS_PER_BLOCK + blob_sidecar.index
+    gindex = BLOB_KZG_COMMITMENT_GINDEX + commitment_item_gindex << floorlog2(BLOB_ZKG_COMMITMENT_GINDEX)
+    return is_valid_merkle_path(
+        leaf=blob_sidecar.kzg_commitment.hash_tree_root(),
+        branch=blob_sidecar.commitment_inclusion_proof,
+        gindex=gindex,
+        root=blob_sidecar.signed_block_header.message.body_root,
+    )
+```
+
+#### `is_valid_merkle_path`
+
+```python
+def is_valid_merkle_path(leaf: Bytes32, branch: Sequence[Bytes32], gindex: int, root: Root) -> bool:
```

Comment:
Does it make sense to reuse [`is_valid_merkle_branch`](https://github.com/ethereum/consensus-specs/pull/3531/files#is_valid_merkle_branch)?

Reply:
We can't re-use that function since this merkle proof involves two nested trees

Reply:
I disagree on that, the validation algo is the same with the diff that a gindex value instead of a regular index is passed onto the call